sd-rsyslog

```diff
@@ -74,15 +74,33 @@ def onInit():
     # emitted you must set 'level' to logging.DEBUG above.)
     logging.debug("onInit called")
 
-
     global process
     if not os.path.exists("/etc/sd-rsyslog.conf"):
         print("Please create the configuration file at /etc/sd-rsyslog.conf", file=sys.stderr)
         sys.exit(1)
     config = configparser.ConfigParser()
     config.read('/etc/sd-rsyslog.conf')
     logvmname = config['sd-rsyslog']['remotevm']
-    localvmname = config['sd-rsyslog'].get('localvm', gethostname())
+    localvmname = config['sd-rsyslog'].get('localvm', None)
+
+    # If no localvm name is specified, it must be supplied by Qubes OS. If this
+    # fails, we exit, to avoid falsely identified logs.
+    if localvmname is None:
+        try:
+            get_vm_name_process = Popen(["/usr/bin/qubesdb-read", "/name"],
+                                        stdout=PIPE, stderr=PIPE)
+            vm_name_output, vm_name_error = get_vm_name_process.communicate()
+            if vm_name_error != b"":
+                print("Error obtaining VM name via qubesdb-read:")
+                print(vm_name_error.decode("utf-8").strip())
+                sys.exit(1)
+            localvmname = vm_name_output.decode("utf-8").strip()
+        except FileNotFoundError:  # not on Qubes?
```

Comment:
Confirmed that `FileNotFoundError` is the proper exception here (same as `subprocess.check_output` would raise):

```
>>> from subprocess import Popen
>>> p = Popen(["/bin/echo2", "hello"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/bin/echo2': '/bin/echo2'
>>> 
```